runtime/src/bank.rs

```diff
@@ -7691,6 +7691,46 @@ impl Bank {
             .epoch_accounts_hash_manager
             .try_get_epoch_accounts_hash()
     }
+
+    /// Write details about this Bank's stored accounts to a file
+    pub fn write_stored_accounts_to_file_for_debugging(&self) {
+        // We're not equipped to compare results on a non-frozen bank
+        if !self.is_frozen() {
+            return;
+        }
+        let slot = self.slot();
+
+        let file_name = format!("{}.{}", slot, self.hash());
+        let path = self
+            .rc
+            .accounts
+            .accounts_db
+            .get_accounts_hash_cache_path()
+            .join(&file_name);
+        if path.exists() {
+            return;
+        }
+
+        let mut accounts = self
+            .rc
+            .accounts
+            .accounts_db
+            .get_stored_accounts_for_slot(slot);
+        // Account pubkeys are the first item in the summary, so we can
+        // just sort based on the entire string for easy diffing
+        accounts.sort();
+
+        use std::fs;
+        // Log the last blockhash incase we have a bad version of the block
+        let last_blockhash = format!("last_blockhash: {}\n", self.last_blockhash());
+        error!(
```

Comment:
probably not 'error'?

Reply:
Agreed - I contemplated stripping it out altogether, but landed on `info!` given that our duplicate check occurs before the log line. Ie, this won't be another line that is repeated 1000's of times when the validator gets stuck.